js/components/multi-audio/multiAudioPopover.js

```diff
@@ -0,0 +1,62 @@
+var React = require('react'),
+  CONSTANTS = require('../../constants/constants'),
+  AccessibleMenu = require('../higher-order/accessibleMenu'),
+  CloseButton = require('../closeButton');
+
+var MultiAudioPopover = React.createClass({
+  /**
+   * close the popover
+   */
+  handleClose: function() {
+    this.props.togglePopoverAction({
+      restoreToggleButtonFocus: true
+    });
+  },
+
+  /**
+   * Call setCurrentAudio from controller
+   * @param {String} id - id of the audio track to activate
+   */
+  handleMultiAudioClick: function(id) {
+    //change audio only if try to set not current audio track
+    if (this.props.controller && this.props.controller.state.currentAudioId !== id) {
```

Comment:
Should we move this check into the ``setCurrentAudio`` function in the controller? I feel like it would be better for the controller to check its own state and have ``multiAudioPopover`` be less dependent on the controllers state.

Reply:
\+